### ISSUE TITLE
MS SQL Server UI is missing DriverUIContributor classes for MS Sql Server 2012 and up

### ISSUE THREAD

**Issue by mrfeet** [eclipse-datatools/datatools#21]
The plugin.xml of org.eclipse.datatools.enablement.msft.sqlserver.ui specifies a driverUIContributor for the following templates:
- org.eclipse.datatools.enablement.msft.sqlserver.2016.driverTemplate
- org.eclipse.datatools.enablement.msft.sqlserver.2014.driverTemplate
- org.eclipse.datatools.enablement.msft.sqlserver.2012.driverTemplate

For example, the referenced class for the 2016 driver is: org.eclipse.datatools.enablement.msft.sqlserver.internal.ui.connection.drivers.SQLServer2012DriverUIContributor

But none of the referenced classes exist in the plugin. 

As a result creating a connection for a MS Sql Server 2016 DB doesn't work. The dialog will not show any properties, and the wizard cannot be finished

**Comment by mrfeet** [comment]
![image](https://github.com/user-attachments/assets/ad45a310-f1a3-4688-9dbc-8cd39cf06bde)

I have tested with the latest eclipse (2024-6) and datatools (1.16.1)
![image](https://github.com/user-attachments/assets/5eb301cd-a729-44ea-8df2-19f8d2158d0d)

**Comment by mrfeet** [comment]
The log is not showing any problem here, because the code that instantiates the DriverUIContributor doesn't log anything when there are problems:

DriverUIContributorRegistry, method initializeDriverCollection():

```
					IConfigurationElement configElement = configElements[elementIndex];
					String driverTemplateID = configElement
							.getAttribute("driverTemplateID"); //$NON-NLS-1$
					IDriverUIContributor driverUIContributor = null;
					try {
						driverUIContributor = (IDriverUIContributor) configElement
								.createExecutableExtension("class"); //$NON-NLS-1$
					} catch (Exception e) {

                                             NOTHING LOGGED HERE!!!!

					}
					if (driverUIContributor != null) {
						driverUIContributorTable.put(driverTemplateID,
								driverUIContributor);
					}

```

**Comment by merks** [comment]
The project is kind of on life support, so if something needs to be fixed, someone needs to volunteer to do that.   I don't personally have the skills to know how to fix a problem like this.  But I can review pull requests:

https://github.com/eclipse-datatools/.github/blob/main/CONTRIBUTING.md

**Comment by tillc24-byte** [comment]
A pretty large gap has occurred in the Microsoft SQL Server implementation with what it supports and the present versions of SQL Server.   Generally, this plugin should probably be updated every one or two years OR the implementation should be slightly more flexible.  In looking at some of the code, and comparing differences between SQL Server versions, I'm not sure I see a need to be so strict on the version handling.  Basically what this code did is end its own useful life.

For instance, the following SQLServerDatabaseRecognizer.java (path below) only recognizes versions that are long EOL.  As of this moment, versions 16 and 17 are out, but this class does not recognize.   If you look at the previous commit for this class, where they added support for one of the versions, someone should probably run through that cycle including the new.  

/plugins/enablement/org.eclipse.datatools.enablement.msft.sqlserver/src/org/eclipse/datatools/enablement/msft/internal/sqlserver/connection/SQLServerDatabaseRecognizer.java


Basically, follow:   https://github.com/eclipse-datatools/datatools/commit/7ce7ca592b4da651d0062fa3943f9ce8ddc6d30b

**Comment by tillc24-byte** [comment]
[datatools-msft-SQLServer 2017 through 2025 support.zip](https://github.com/user-attachments/files/26277546/datatools-msft-SQLServer.2017.through.2025.support.zip)

Uploading the attached if it helps anyone else.  I took the git project and forked the 1.3.0 and 1.4.0 Microsoft Implementation - and added support for SQL Server 2017, 2019, 2022, and 2025.  I also fixed some of the handling where schemas and tables were not presented because of handling of the catalog name (null or master versus specified).  In testing my changes, the only defect that I did not fix was within DSE, if you connected to a different catalog/database, the schemas and tables presented are for the catalog/database which you were connected to.  As this required larger changes and is not a show stopper, I ignored.   I also enhanced the connection dialog for 2022 and 2025 to include connection properties such as trust server certificate.

<img width="1638" height="391" alt="Image" src="https://github.com/user-attachments/assets/b9b1d11d-ee40-4eea-bcad-d655368d144f" />

<img width="737" height="508" alt="Image" src="https://github.com/user-attachments/assets/58bbb697-b37d-45f7-8926-7f06fc22b94b" />

<img width="900" height="640" alt="Image" src="https://github.com/user-attachments/assets/920dc6df-2f88-43c4-b3bd-5d1320c80433" />




To deploy the plugin (assuming it's deployed as attached), here's the process.

1) Upgrade your Eclipse version. I tested on 2026-03 although given the age of the MS SQL Server plugin, I suspect things will work on some of the older versions but without testing, support is unknown.

2) Take the three attached files (3 jars - not with the timestamp - not SNAPSHOT) and place under the "\eclipse\plugins" of your eclipse installation.  After placing, that location will have the old plugs and new plugins.  Personally I see no harm in that as Eclipse uses a registry to maintain plugin locations.

3) Navigate open the file under your eclipse configuration.  You should probably save a copy of this file prior to editing. "eclipse\configuration\org.eclipse.equinox.simpleconfigurator\bundles.info"


4) Within the bundles.info file, there's three lines we need to comment out and three lines we need to add.  To register the updated plugin, we're manually registering the plugin versions and their file locations.  Just note the newer 1.3.1 or 1.4.1 changes with the generated timestamps.  Make the below changes and then save this file.  

# Add in the new

org.eclipse.datatools.enablement.msft.sqlserver,1.4.1.202603260726,plugins/org.eclipse.datatools.enablement.msft.sqlserver_1.4.1.202603260726.jar,4,false
org.eclipse.datatools.enablement.msft.sqlserver.dbdefinition,1.3.1.202603260726,plugins/org.eclipse.datatools.enablement.msft.sqlserver.dbdefinition_1.3.1.202603260726.jar,4,false
org.eclipse.datatools.enablement.msft.sqlserver.ui,1.3.1.202603260726,plugins/org.eclipse.datatools.enablement.msft.sqlserver.ui_1.3.1.202603260726.jar,4,false
# Comment out the old
#org.eclipse.datatools.enablement.msft.sqlserver,1.4.0.202311071307,plugins/org.eclipse.datatools.enablement.msft.sqlserver_1.4.0.202311071307.jar,4,false
#org.eclipse.datatools.enablement.msft.sqlserver.dbdefinition,1.3.0.202412071309,plugins/org.eclipse.datatools.enablement.msft.sqlserver.dbdefinition_1.3.0.202412071309.jar,4,false
#org.eclipse.datatools.enablement.msft.sqlserver.ui,1.3.0.202311071249,plugins/org.eclipse.datatools.enablement.msft.sqlserver.ui_1.3.0.202311071249.jar,4,false


Good luck.. Hope it helps someone else or inspires someone to actually fix.

**Comment by tillc24-byte** [comment]
> ![image](https://github.com/user-attachments/assets/ad45a310-f1a3-4688-9dbc-8cd39cf06bde)
> 
> I have tested with the latest eclipse (2024-6) and datatools (1.16.1) ![image](https://github.com/user-attachments/assets/5eb301cd-a729-44ea-8df2-19f8d2158d0d)

I can confirm that there was a missing configuration (SQLServer2016DriverUIContributor) in the 2016 SQL Server support which prevents the UI from loading the driver properties.
